pylabrobot/centrifuge/agilent_centrifuge.py

```diff
@@ -0,0 +1,399 @@
+    if not self.dev:
+      raise RuntimeError("Device not initialized")
+
+    logger.debug("Sending %s", cmd.hex())
+    written = self.dev.write(cmd.decode("latin-1"))
+    logger.debug("Wrote %s bytes", written)
+
+    if written != len(cmd):
+      raise RuntimeError("Failed to write all bytes")
+    return await self.read_resp(timeout=read_timeout)
+
+  async def send_payloads(self, payloads) -> None:
+    """Send a list of commands to the centrifuge."""
+    for tx in payloads:
+      if isinstance(tx, str):
+        byte_literal = bytes.fromhex(tx)
+        await self.send(byte_literal)
+      else:
+        await self.send(tx)
+
+# setup() and stop() helper functions: configure_and_initialize, set_configuration_data, initialize
+
+  async def configure_and_initialize(self):
+    await self.set_configuration_data()
+    await self.initialize()
+
+  async def set_configuration_data(self):
+    """Set the device configuration data."""
+    self.dev.ftdi_fn.ftdi_set_latency_timer(16)
+    self.dev.ftdi_fn.ftdi_set_line_property(8, 1, 0)
+    self.dev.ftdi_fn.ftdi_setflowctrl(0)
+    self.dev.baudrate = 19200
+
+  async def initialize(self):
+    """Initialize the device."""
+    self.dev.write(b"\x00" * 20)
+    for i in range(33):
+      packet = b"\xaa" + bytes([i & 0xFF, 0x0e, 0x0e + (i & 0xFF)]) + b"\x00" * 8
+      self.dev.write(packet)
+    await self.send(b"\xaa\xff\x0f\x0e")
+
+# Centrifuge operations
+
+  async def open_door(self):
+    await self.send(b"\xaa\x02\x26\x00\x07\x2f")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+  async def close_door(self):
+    await self.send(b"\xaa\x02\x26\x00\x05\x2d")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+  async def lock_door(self):
+    await self.send(b"\xaa\x02\x26\x00\x01\x29")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+  async def unlock_door(self):
+    await self.send(b"\xaa\x02\x26\x00\x05\x2d")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+  async def lock_bucket(self):
+    await self.send(b"\xaa\x02\x26\x00\x07\x2f")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+  async def unlock_bucket(self):
+    await self.send(b"\xaa\x02\x26\x00\x06\x2e")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+  async def go_to_bucket1(self):
+    await self.get_status()
+    new_position = (self.position + 4000).to_bytes(4, byteorder="little")
+    byte_string = b"\xaa\x01\xd4\x97" + new_position + b"\xc3\xf5\x28\x00\xd7\x1a\x00\x00"
+    sum_byte = (sum(byte_string)-0xaa)&0xff
+    byte_string += sum_byte.to_bytes(1, byteorder="little")
+
+    move_bucket = [
+  "aa 02 26 00 00 28",
+  "aa 02 0e 10",
+  "aa 01 17 02 1a",
+  "aa 01 0e 0f",
+  "aa 01 e6 c8 00 b0 04 96 00 0f 00 4b 00 a0 0f 05 00 07",
+  "aa 01 17 04 1c",
+  "aa 01 17 01 19",
+  "aa 01 0b 0c",
+  "aa 01 e6 c8 00 b0 04 96 00 0f 00 4b 00 a0 0f 05 00 07",
+  byte_string
+  ]
+    await self.close_door()
+    await self.lock_door()
+
+    await self.send_payloads(move_bucket)
+
+    time.sleep(2)
```

Comment:
```suggestion
    await asyncio.sleep(2)
```
otherwise you will block the event loop